scripts/core/CONTRIB.rst

```diff
@@ -1,71 +1,204 @@
+in the repository as a result of `Generating Source`_.
+
+Before your pull request is merged it *must* pass all jobs in the GitHub
+Actions workflow and *must* be reviewed by no less than two code owners.
+
+.. hint::
+
+    When rebasing a branch on top of ``main`` results in merged conflicts it is
+    recommended to resolve conflicts in the ``*.yml`` files then `Generating
+    Source`_. This will automatically resolve conflicts in the generated source
+    files, leaving only conflicts in non-generated source files to be resolved,
+    if any.
+
+Core Features
+=============
+
+A core feature *must* have a stable API/ABI and *should* strive to be supported
+across all adapters. However, core features *may* be optional and thus only
+supported in one or more adapters. A core feature *should* also strive to
+enable similar functionality in parallel language runtime (such as SYCL,
+OpenMP, ...) where possible although this is a secondary concern.
+
+.. hint::
+    Optional features should be avoided as much as possible to maximize
+    portability across adapters and reducing the overhead required to use make
+    use of features in parallel language runtimes.
+
+Core features are defined in the ``*.yml`` files in the `scripts/core
+<https://github.com/oneapi-src/unified-runtime/tree/main/scripts/core>`_
+directory. Most of the files are named after the API object who's interface is
+defined within them, with the following exceptions:
+
+*   `scripts/core/common.yml`_ defines symbols which are used by multiple
+    interfaces through the specification, e.g. macros, object handles, result
+    enumerations, and structure type enumerations.
+*   `scripts/core/enqueue.yml`_ defines commands which can be enqueued on a
+    queue object.
+*   `scripts/core/runtime.yml`_ defines global symbols pertaining to
+    initialization and tear down of the entire runtime.
+*   `scripts/core/registry.yml`_ contains an enumeration of all entry-points,
+    past and present, for use in the XPTI tracing framework. It is
+    automatically updated so shouldn't require manual editing.
+*   ``scripts/core/exp-<feature>.yml`` see `Experimental Features`_.
+
+.. _scripts/core/common.yml:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/common.yml
+.. _scripts/core/enqueue.yml:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/enqueue.yml
+.. _scripts/core/runtime.yml:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/runtime.yml
+.. _scripts/core/registry.yml:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/core/registry.yml
+
+Core Optional Features
+----------------------
+
+Optional core features *must* be supported in at least one adapter. Support for
+an optional core feature *must* be programmatically exposed to the user via
+boolean query call to ${x}DeviceGetInfo and a new enumerator of the form
```

Comment:
Should this be
```suggestion
boolean query call to ``${x}DeviceGetInfo`` and a new enumerator of the form
```
?

Reply:
Unfortunately these are run through some scripts the replace `${x}` with a cross-reference that breaks when wrapped in inline monospace markup otherwise I would have done this.